src/adapters/fileSystem.ts

```diff
@@ -0,0 +1,23 @@
+import * as fs from 'fs'
+import IFileSystem from '../ports/IFileSystem'
+
+export function createFileManager(contentDirectory: string): IFileSystem {
+  function saveFile(file: string, data: ArrayBuffer) {
+    fs.writeFileSync(`${contentDirectory}/${file}`, Buffer.from(data))
+    return `${contentDirectory}/${file}`
+  }
+
+  function retrieveFile(file: string): Buffer {
+    return fs.readFileSync(`${contentDirectory}/${file}`)
+  }
+
+  function asReadable(file: string): fs.ReadStream {
+    return fs.createReadStream(`${contentDirectory}/${file}`)
+  }
+
+  function deleteFile(file: string): void {
+    return fs.unlinkSync(`${contentDirectory}/${file}`)
```

Comment:
Always prefer async versions of functions in servers, it will block the whole process and other requests otherwise
```suggestion
    try {
      await fsPromises.unlink(`${contentDirectory}/${file}`)
    } catch { /* LOG */ }
```

Same for `readFile`